File: Java Spring Boot Project/misc/nport_project/MtCProject1/src/main/java/com/nikosportolos/MtCProject1/models/responses/AllUnitsResponse.java
package com.nikosportolos.MtCProject1.models.responses;

import com.nikosportolos.MtCProject1.models.Unit;

import java.util.List;

public class AllUnitsResponse {

    private List<UnitResponse> units;

    public AllUnitsResponse(List<UnitResponse> units) {
        this.units = units;
    }

    public List<UnitResponse> getUnits() {
        return units;
    }

    public void setUnits(List<UnitResponse> units) {
        this.units = units;
    }
}

File: Java Spring Boot Project/company_directory_app/lib/models/responses/all_units_response.dart
import 'package:company_directory_app/models/responses/unit_response.dart';

class AllUnitsResponse {
  final List<UnitResponse> units;

  AllUnitsResponse({this.units});

  factory AllUnitsResponse.fromJson(Map<String, dynamic> json) {
    var list = json['units'] as List;
    List<UnitResponse> itemsList = list.map((i) => UnitResponse.fromJson(i)).toList();

    return AllUnitsResponse(units: itemsList);
  }
}
